assembly/src/parsers/mod.rs

```diff
@@ -316,6 +349,47 @@ fn parse_param<I: core::str::FromStr>(op: &Token, param_idx: usize) -> Result<I,
     Ok(result)
 }
 
+/// parses a constant token and returns a (constant_name, constant_value) tuple
+pub fn parse_constant(token: &Token) -> Result<(String, u64), ParsingError> {
+    match token.num_parts() {
+        0 => unreachable!(),
+        1 => Err(ParsingError::missing_param(token)),
+        2 => {
+            let const_declaration: Vec<&str> = token.parts()[1].split('=').collect();
+            match const_declaration.len() {
+                0 => unreachable!(),
+                1 => Err(ParsingError::missing_param(token)),
+                2 => {
+                    let name = CONSTANT_LABEL_PARSER
+                        .parse_label(const_declaration[0].to_string())
+                        .map_err(|err| ParsingError::invalid_const_name(token, err))?;
+                    let value = parse_const_value(token, const_declaration[1], 0..Felt::MODULUS)?;
+                    Ok((name, value))
+                }
+                _ => Err(ParsingError::extra_param(token)),
+            }
+        }
+        _ => Err(ParsingError::extra_param(token)),
+    }
+}
+
+/// Parses a constant value and ensures it falls within bounds specified by the caller
+fn parse_const_value<R: RangeBounds<u64>>(
+    op: &Token,
+    const_value: &str,
+    range: R,
+) -> Result<u64, ParsingError> {
+    let result = const_value
+        .parse::<u64>()
+        .map_err(|err| ParsingError::invalid_const_value(op, const_value, &err.to_string()))?;
+
+    range.contains(&result).then_some(result).ok_or_else(|| ParsingError::invalid_const_value(op, const_value, format!(
+        "constant value must be greater than or equal to {lower_bound} and less than or equal to {upper_bound}", lower_bound = bound_into_included_u64(range.start_bound(), true), 
+        upper_bound = bound_into_included_u64(range.end_bound(), false)
+    )
+    .as_str(),))
+}
```

Comment:
Since this is a specialized function for parsing constant values, should we just define lower/upper bounds in the function rather than passing them in as parameters?

Reply:
My thinking was that maybe in the future we will be able to specify typed constants.  e.g. `const.MY_U32.1234.u32` which would validate that the constant is in fact u32 and falls within appropriate range.  However seeing as that's outside the scope of this PR it's probably best to remove it.